inst/include/dust/dust.hpp

```diff
@@ -32,48 +33,43 @@ class Dust {
 
   Dust(const pars_t& pars, const size_t step, const size_t n_particles,
        const size_t n_threads, const std::vector<uint64_t>& seed,
-       size_t device_id) :
+       cuda::device_config device_config) :
     n_pars_(0),
     n_particles_each_(n_particles),
     n_particles_total_(n_particles),
     pars_are_shared_(true),
     n_threads_(n_threads),
-    device_id_(device_id),
+    device_config_(device_config),
     rng_(n_particles_total_ + 1, seed),  // +1 for the filter's rng state
     errors_(n_particles_total_),
     stale_host_(false),
     stale_device_(true),
     select_needed_(true),
-    select_scatter_(false),
-    shared_size_(0) {
-#ifdef __NVCC__
-    initialise_device(device_id);
-#endif
+    select_scatter_(false) {
     initialise(pars, step, n_particles, true);
     initialise_index();
     shape_ = {n_particles};
+#ifdef DUST_USING_CUDA_PROFILER
+    cuda_profiler_start(device_config_);
+#endif
   }
 
   Dust(const std::vector<pars_t>& pars, const size_t step,
        const size_t n_particles, const size_t n_threads,
-       const std::vector<uint64_t>& seed, size_t device_id,
+       const std::vector<uint64_t>& seed, cuda::device_config device_config,
```

Comment:
```suggestion
       const std::vector<uint64_t>& seed, cuda::device_config& device_config,
```

Reply:
This won't change the number of copies (we could pass it as `const cuda::device_config&` but at some point it will get copied I belive